File: demo/OpenHospital/baseline/backend/schemas/examination.py
"""Examination schemas."""

from typing import List, Optional, Dict, Any, Literal
from pydantic import BaseModel


class ExaminationResult(BaseModel):
    """Single examination result."""
    name: str
    result: str
    reference_range: Optional[str] = None
    abnormal: Optional[bool] = None


class ExaminationRecord(BaseModel):
    """Examination record."""
    id: str
    patient_id: str
    doctor_id: str
    examination_items: List[str]
    ordered_tick: int
    status: Literal["pending", "completed"]
    completed_tick: Optional[int] = None
    results: Optional[Dict[str, Any]] = None

File: demo/OpenHospital/baseline/frontend/src/types/examination.ts
export interface ExaminationRecord {
  id: string
  patient_id: string
  doctor_id: string
  examination_items: string[]   // e.g., ["血常规", "心电图", "胸部CT"]
  ordered_tick: number
  status: ExaminationStatus
  completed_tick?: number
  results?: Record<string, ExaminationResult>
}

export type ExaminationStatus = 'pending' | 'completed'

export interface ExaminationResult {
  name: string
  result: string
  reference_range?: string
  abnormal?: boolean
}

